### PR TITLE
Enable logs with stacktraces

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What does this PR do?

Enable to send logs with stacktraces.

### Additional Notes

Unminification on:

**RUM (iOS/Android)**
image
image

**Logs (iOS/Android)**
image
image

**Error tracking (iOS/Android)**
Note: The "unknown language" is being worked on by the Error Tracking team.

image
image

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [x] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

File: packages/core/android/src/main/kotlin/com/datadog/reactnative/DdLogs.kt [hunk 1]
```diff
@@ -84,4 +85,120 @@ class DdLogs(reactContext: ReactApplicationContext, logger: Logger? = null) :
         )
         promise.resolve(null)
     }
+
+    /**
+     * Send a log containing an error with level debug.
```
Comment (0xnm) [inline packages/core/android/src/main/kotlin/com/datadog/reactnative/DdLogs.kt:90]: minor grammar (same for other methods):

```suggestion
     * Send a log containing an error with Debug level.
```

File: packages/core/android/src/main/kotlin/com/datadog/reactnative/DdLogs.kt [hunk 2]
```diff
@@ -84,4 +85,120 @@ class DdLogs(reactContext: ReactApplicationContext, logger: Logger? = null) :
...
+            errorKind = errorKind,
+            errorMessage = errorMessage,
+            errorStacktrace = stacktrace,
+            attributes = context.toHashMap() + GlobalState.globalAttributes
+        )
+        promise.resolve(null)
+    }
+
+    /**
+     * Send a log with containing an error level info.
```
Comment (0xnm) [inline packages/core/android/src/main/kotlin/com/datadog/reactnative/DdLogs.kt:119]:
```suggestion
     * Send a log with containing an error with Info level.
```

File: packages/core/android/src/test/kotlin/com/datadog/reactnative/DdLogsTest.kt [hunk 3]
```diff
@@ -41,6 +43,12 @@ internal class DdLogsTest {
     @StringForgery
     lateinit var fakeMessage: String
 
+    private var fakeErrorKind: String? = null
+
+    private var fakeErrorMessage: String? = null
+
+    private var fakeStacktrace: String? = null
```
Comment (0xnm) [inline packages/core/android/src/test/kotlin/com/datadog/reactnative/DdLogsTest.kt:50]:
```suggestion
    private lateinit var fakeErrorKind: String?

    private lateinit var fakeErrorMessage: String?

    private lateinit var fakeStacktrace: String?
```

File: packages/core/src/logs/types.ts [hunk 4]
```diff
@@ -15,28 +15,28 @@ export type DdLogsType = {
      * @param message: The message to send.
```
Comment (0xnm) [inline packages/core/src/logs/types.ts:15]: should we update doc?

File: packages/core/src/nativeModulesTypes.ts [hunk 5]
```diff
@@ -100,6 +100,70 @@ export type DdNativeLogsType = {
...
+    debugWithError(
+        message: string,
+        errorKind?: string,
+        errorMessage?: string,
+        stacktrace?: string,
+        context?: object
+    ): Promise<void>;
+
+    /**
+     * Send a log with level debug.
```
Comment (0xnm) [inline packages/core/src/nativeModulesTypes.ts:121]:
```suggestion
     * Send a log with INFO level.
```

**Review by 0xnm — COMMENTED** [review]
overall lgtm, but I added some questions/suggestions, mostly related to the documentation though.

**Review by maycmlee — APPROVED** [review]

**Pull request merged**